src/sidebar/test/cross-origin-rpc-test.js

```diff
@@ -212,5 +203,71 @@ describe('crossOriginRPC', function() {
         );
       });
     });
+
+    describe('pre-start requests', function() {
+      beforeEach(function() {
+        crossOriginRPC.server.preStart(fakeWindow);
+      });
+
+      it('responds to an incoming request that arrives before the server starts', function() {
+        fakeWindow.emitter.emit('message', {
+          data: { jsonrpc: '2.0', method: 'changeFocusModeUser', id: 42 },
+          origin: 'https://allowed1.com',
+          source: frame,
+        });
+        crossOriginRPC.server.start(fakeStore, settings, fakeWindow);
+        assert.isTrue(
+          frame.postMessage.calledWithExactly(
+            {
+              jsonrpc: '2.0',
+              id: 42,
+              result: 'ok',
+            },
+            'https://allowed1.com'
+          )
+        );
+      });
+
+      it('responds to multiple incoming requests that arrive before the server starts', function() {
+        const payload = id => ({
```

Comment:
Maybe `messageEvent` might be slightly clearer than `payload` in terms of what kind of object the result is supposed to mimic?